extras/indexing/src/main/java/org/apache/rya/api/client/accumulo/AccumuloExecuteSparqlQuery.java

```diff
@@ -0,0 +1,150 @@
+import edu.umd.cs.findbugs.annotations.NonNull;
+
+/**
+ * An Accumulo implementation of the {@link LoadStatementsFile} command.
+ */
+@DefaultAnnotation(NonNull.class)
+public class AccumuloExecuteSparqlQuery extends AccumuloCommand implements ExecuteSparqlQuery {
+    private static final Logger log = Logger.getLogger(AccumuloExecuteSparqlQuery.class);
+
+    private final InstanceExists instanceExists;
+
+    /**
+     * Constructs an instance of {@link AccumuloExecuteSparqlQuery}.
+     *
+     * @param connectionDetails - Details about the values that were used to create
+     *   the connector to the cluster. (not null)
+     * @param connector - Provides programmatic access to the instance of Accumulo
+     *   that hosts Rya instance. (not null)
+     */
+    public AccumuloExecuteSparqlQuery(final AccumuloConnectionDetails connectionDetails, final Connector connector) {
+        super(connectionDetails, connector);
+        instanceExists = new AccumuloInstanceExists(connectionDetails, connector);
+    }
+
+
+    @Override
+    public String executeSparqlQuery(final String ryaInstanceName, final String sparqlQuery)
+            throws InstanceDoesNotExistException, RyaClientException {
+        requireNonNull(ryaInstanceName);
+        requireNonNull(sparqlQuery);
+
+        // Ensure the Rya Instance exists.
+        if(!instanceExists.exists(ryaInstanceName)) {
+            throw new InstanceDoesNotExistException(String.format("There is no Rya instance named '%s'.", ryaInstanceName));
+        }
+
+
+        Sail sail = null;
+        SailRepository sailRepo = null;
+        SailRepositoryConnection sailRepoConn = null;
+
+        try {
+            // Get a Sail object that is connected to the Rya instance.
+            final AccumuloRdfConfiguration ryaConf = getAccumuloConnectionDetails().buildAccumuloRdfConfiguration(ryaInstanceName);
+            sail = RyaSailFactory.getInstance(ryaConf);
+
+            // Load the file.
+            sailRepo = new SailRepository(sail);
+            sailRepoConn = sailRepo.getConnection();
+            final long start = System.currentTimeMillis();
+            final TupleQuery tupleQuery = sailRepoConn.prepareTupleQuery(QueryLanguage.SPARQL, sparqlQuery);
+            final TupleQueryResult result = tupleQuery.evaluate();
+            final StringBuilder sb = new StringBuilder();
+            final String newline = "\n";
+            sb.append("Query Result:").append(newline);
+            sb.append("Binding Names: ").append(result.getBindingNames().toString()).append(newline);
+
+            int count = 0;
+            while(result.hasNext()) {
+                sb.append(result.next().toString()).append(newline);
+                count++;
+            }
+            final String seconds = new DecimalFormat("0.0##").format((System.currentTimeMillis() - start) / 1000.0);
+            sb.append("Retrieved ").append(count).append(" results in ").append(seconds).append(" seconds.");
```

Comment:
I think it is appropriate.  The contract specifies that this method will return "A user-friendly String representation of the query results."  If you use other query shells, like impala, they also provide query execution timing.  It provides basic feedback to the user for how performant their query is.  If you prefer, we could move the timing section to rya.console and append it to the content that is returned from this method.  Let me know which you prefer, but I really like seeing timing when I issue a query.